### PR TITLE
Fixes #4629 - Patternfly CSS and fonts

### PR DESCRIPTION
**Pull request by dLobatog**
To do:
- update to the latest Patternfly version
- consider using [font-awesome-rails](https://github.com/bokmann/font-awesome-rails)
- look into using a more patternflied [dashboard](https://www.patternfly.org/wp-content/uploads/patternfly/tests/dashboard.html) and trends

### CODE REVIEW

**Comment by ares** [comment]
@eLobato please fix failing tests

File: app/helpers/layout_helper.rb [hunk 1]
```diff
@@ -294,14 +294,28 @@ def alert(opts = {})
     content_tag :div, :class => html_class do
       result = "".html_safe
       result += alert_close if opts[:close]
-      result += alert_header(opts[:header])
-      result += opts[:text].html_safe
+      result += alert_header(opts[:header], opts[:class])
+      result += "#{opts[:text]}".html_safe
```
Comment (ShimShtein) [inline app/helpers/layout_helper.rb]: nitpick: Shouldn't it be `result += opts[:text].to_s.html_safe`?

File: app/views/hosts/show.html.erb [hunk 2]
```diff
@@ -38,10 +38,7 @@
     </ul>
     <div id="myTabContent" class="tab-content">
       <div class="tab-pane active in" id="properties" data-ajax-url='<%= overview_host_path(@host)%>'>
-        <p id="spinner">
-          <%= image_tag 'spinner.gif' %>
-          <%= _('Loading host information ...') %>
-        </p>
+        <%= spinner('Loading host information ...') %>
```
Comment (ShimShtein) [inline app/views/hosts/show.html.erb]: What about translation? `_('...')`
It's the same comment for the whole file

File: app/views/roles/_form.html.erb [hunk 3]
```diff
@@ -20,10 +20,7 @@
         </div>
       <% else %>
         <div data-ajax-url="<%= filters_path(:role_id => @role) %> table">
-          <p id="spinner">
-            <%= image_tag 'spinner.gif' %>
-            <%= _('Loading filters...') %>
-          </p>
+          <%= spinner('Loading filters ...') %>
```
Comment (ShimShtein) [inline app/views/roles/_form.html.erb]: Translation?

File: app/assets/stylesheets/bootstrap_and_overrides.scss [hunk 4]
```diff
@@ -3,10 +3,27 @@
 
 $navbar-default-bg: $primary_color;
 
-@import "bootstrap";
+@import "patternfly";
 
 .dropdown-menu {
-  border-radius: 0 0 6px 6px;
+  border-radius: 0, 0, 6px, 6px;
```
Comment (tbrisker) [inline app/assets/stylesheets/bootstrap_and_overrides.scss]: why the commas?

File: vendor/assets/stylesheets/multi-select.scss [hunk 5]
```diff
@@ -100,27 +97,22 @@
...
-    -moz-border-radius: 4px 4px 0 0;
-    border-radius: 4px 4px 0 0;
+  padding: 4px 10px;
+  border: 1px solid #d1d1d1;
+  background-clip: padding-box;
+  background-color: #f9f9f9;
+  background-image: -webkit-linear-gradient(top, #fafafa 0%, #ededed 100%);
+  background-image: linear-gradient(to bottom, #fafafa 0%, #ededed 100%);
+  background-repeat: repeat-x;
+  filter: progid:DXImageTransform.Microsoft.gradient(startColorstr='#fffafafa', endColorstr='#ffededed', GradientType=0);
```
Comment (tbrisker) [inline vendor/assets/stylesheets/multi-select.scss]: we have a mixin for linear-gradiant

File: app/views/config_templates/_form.html.erb [hunk 6]
```diff
@@ -88,8 +88,7 @@
         <p id="snippet_message" class="alert alert-info" <%= display? !@config_template.snippet %> ><%= _("Not relevant for snippet") %></p>
 
         <div id="association" <%= display? @config_template.snippet %> >
-          <div class="alert alert-success alert-dismissable">
-            <%= alert_close %>
+          <%= alert :class=>"alert-info", :close=>true, :header => _("How templates are determined"), :text => association_text %>
```
Comment (tbrisker) [inline app/views/config_templates/_form.html.erb]: undefined local variable or method `association_text'

File: app/views/subnets/index.html.erb [hunk 7]
```diff
@@ -8,7 +8,7 @@
     <tr>
       <th><%= sort :name, :as => s_("Subnet|Name") %></th>
       <th><%= sort :network, :as => s_("Subnet|Network") %></th>
-      <th><%= _("Domains") %></th>
+      <th><%= sort :domains, :as => _("Domains") %></th>
```
Comment (tbrisker) [inline app/views/subnets/index.html.erb]: Invalid search query: the field 'domains' in the order statement is not valid field for search

File: app/helpers/layout_helper.rb (hunk 1 above)
Comment (dLobatog) [inline app/helpers/layout_helper.rb]: Interpolation `#{string}` performs `.to_s` on the object you call it on, in this case `opts[:text]`

File: vendor/assets/stylesheets/multi-select.scss (hunk 5 above)
Comment (dLobatog) [inline vendor/assets/stylesheets/multi-select.scss]: Not sure how can I use it for files in `vendor`?

**Comment by dLobatog** [comment]
@ares @tbrisker @ShimShtein I tried to address your comments, hope tests pass now

File: vendor/assets/stylesheets/multi-select.scss (hunk 5 above)
Comment (tbrisker) [inline vendor/assets/stylesheets/multi-select.scss]: just put `@import "mixin";` at the top. 
While you're at it, we also have mixin for box-shadow, and transition no longer needs all the browser prefixes (http://caniuse.com/#feat=css-transitions) for any modern browser

File: app/assets/stylesheets/bootstrap_and_overrides.scss [hunk 8]
```diff
@@ -3,12 +3,29 @@
 
 $navbar-default-bg: $primary_color;
 
-@import "bootstrap";
+@import "patternfly";
```
Comment (tbrisker) [inline app/assets/stylesheets/bootstrap_and_overrides.scss:6]: wouldn't it make sense to rename this file to `patternfly_and_overrides.scss`?

File: app/assets/stylesheets/bootstrap_and_overrides.scss [hunk 9]
```diff
@@ -3,12 +3,29 @@
...
+
+.autocomplete-clear{
+  font-size: 14px
+}
+.autocomplete-input{
+  min-height: 27px;
+}
+
 // Necessary because bootstrap3 nav dropdown does not have top arrow
 .nav li .dropdown-menu:after {
```
Comment (tbrisker) [inline app/assets/stylesheets/bootstrap_and_overrides.scss]: is this also necessary for patternfly? if not we can get rid of it.

**Comment by tbrisker** [comment]
A few general comments:
1. The font is slightly smaller on most pages, making the text difficult to read on high resolution screens.
2. Full width makes the tables look weird. Smaller tables (like hostgroups for example) are stretched and have a lot of empty space all over the place. Bigger tables (like hosts for example) have the text being cut off way too soon leaving blank space which could have been used for text. In all tables the right most column (the buttons) have too much space making them also look awkward.
3. If we go with full width, I would suggest adding a `padding: 0 5%` to the `#main` element to make it a bit less stretched out.

**Comment by dLobatog** [comment]
[test] because of the latest enabled cops

**Comment by dLobatog** [comment]
@kybaker @ohadlevy Any comments? I personally prefer not to add the padding to #main, but don't trust me with design. Font wise, it's Patternfly's default.

**Comment by ohadlevy** [comment]
personally  i prefer fixed width as it is today, full width requires an
update to most pages ATM.

File: app/assets/stylesheets/bootstrap_and_overrides.scss (hunk 8 above)
Comment (dLobatog) [inline app/assets/stylesheets/bootstrap_and_overrides.scss:6]: Yes, changed.

File: app/assets/stylesheets/bootstrap_and_overrides.scss (hunk 9 above)
Comment (dLobatog) [inline app/assets/stylesheets/bootstrap_and_overrides.scss]: I don't think so, I removed that.

**Comment by dLobatog** [comment]
@tbrisker @ohadlevy Thanks for the comments, I updated it with your suggestions (7% padding instead of 5% as it was closer to the current layout). I noticed power buttons and delete buttons are not colored on the Host view.

Also in case you're curious how other paddings look, this is _10%_:




and _full-width_:

**Comment by ohadlevy** [comment]
@kybaker: your thoughts?

**Comment by kybaker** [comment]
@eLobato I put together some feedback on this PF integration on a few page types. The primary inconsistency appears in the navigation.

PatternFly Adoption Issues - 

Navigation
• Typography is currently using the incorrect typeface and styles, the styles should mirror the nav bar examples below.
• Dropdowns are using using incorrect styles and sizing. Typefaces are also using incorrect styles. The styles should mirror the nav bar examples below.
• The load bar on the top is an inconsistent blue, would look better if it used a darker blue from PatternFly.
https://www.patternfly.org/wikis/patterns/navigation/primary-navigation-bar/examples/

Homepage
• Container corners should be square
• The Manage Dashboard arrow should be the same as the bookmark dropdown
• The Manage Dashboard overlaps the container in the smaller responsive state.

New/add button across application
• Using weird/wrong styles, ideally the button should use the primary blue button styles. We have successfully tested the blue button as part of larger workflows. I would recommend the blue.
https://www.patternfly.org/widgets/#buttons

Hosts/Table Page
• New Host button is not using PF styles (see above)
• Table column headers style and sort functions differently than PF. The PF table design and functions have been successfully tested and vetted.
• Edit button uses the wrong arrow. Should use the consistent dropdown arrow (see above)
https://www.patternfly.org/wp-content/uploads/patternfly/tests/basic.html

Host Detail
• Header spacing and buttons is weird. Should use the same container layout as the host table page. 
• Square corners on the content

All Container 
• Spacing is weird on the top
• Search is inconsistent, should be similar to the search in the table pages

**Comment by dLobatog** [comment]
@kybaker Thanks! I'll disregard the container comments for the moment as these are more related with [foreman-docker](https://github.com/theforeman/foreman-docker) than Foreman, I'll let you know when I update the PR with fixes to the rest.

**Comment by dLobatog** [comment]
@ohadlevy @kybaker I updated the PR with
- Patternflied navbar (except for the locs/orgs selector, which I think should be a Patternfly context selector)
- Made most corners square
- Updated most alerts to use patternfly except for the jnotify ones (top-right corner), which should follow.
- Rebased 
- Other minor updates

We miss (apart from what @kybaker comments above):
- Patternfly spinners
- Patternfly data tables
- Patternfly clear search button
- Patternfly jnotify alerts
- Patternfly host provisioning loading bar
- Proper grid usage (I'd save this one for a future PR, too much of a change)

Let me know if you miss anything else, preferably with screenshots.

**Comment by ehelms** [comment]
A few notes from myself after clicking through some pages:
1. Menu links have a 'text-shadow' that makes the font look broken, recommend removal of text-shadow
2. 404s on OpenSans-Bold font on a few pages
3. Why 7% margins left and right? I might suggest a fixed pixel number instead. I think the default is 20px so a multiple of that (40 or 60 looked best if the majority don't like the standard 20px that both bootstrap and patternfly prescribe by default)
4. You are including both patternfly.css and patternfly.scss files? The scss file looking like just a regular css file at first glance.

Other than those points, everything looked good and no clashes with the current Katello pages either. I did not test removal of patternfly from Bastion + this PR but can if you want me to.

Something to note if you are not familiar with the concept, as we have been using it for a while now with our assets is using bower.json to control asset versions and file placement. In Bastion, you can see here (https://github.com/Katello/bastion/blob/master/bower.json#L24) where we are able to note the version and further down https://github.com/Katello/bastion/blob/master/bower.json#L93 not the specific files. With a small customization to a Grunt plugin anytime we want to update, we can just bump the version, run bower update, and then we are able to commit the change fairly easily while keeping track of what we have been using.

**Comment by dLobatog** [comment]
@ehelms I think I've addressed all of your concerns in the latest commit.

@tbrisker @kybaker @ohadlevy Except for alerts and spinners. I think it should be good to go. I like bower but I'd prefer to keep changes minimal, hence why I only changed what was strictly necessary to make it look 'patternflied'.

Let me know if you guys think should be fixed - otherwise I'll only focus on making alerts/spinners and call it done.

Thanks everyone!

**Comment by ehelms** [comment]
@eLobato sort of a nitpick but you are including patternfly as an 'scss' file which would imply (to those that know scss) that it includes all of the mixins/variables/etc that come with the scss version. Can we use the .css extension so its clear that it is only the built CSS version?

**Comment by ehelms** [comment]
I tested removing the Patternfly from Bastion and other than minor structural tweaks everything looked great. Since we also install foreman_docker by default, I perused the UI items from it and only noticed a missing false.png file. On the breadcrumbs for wizards, I was seeing the arrow point appear larger:

new organization

Updating `.wizard li` to `padding: 10px 12px 9px` fixed it for me.

**Comment by dLobatog** [comment]
@ehelms thanks for the comment, fixed!

Latest commit updates Patternfly to the latest version, and Font-awesome to 4.2. It also fixes spinners to use the Font-awesome/patternfly classes instead of the gif file, and a few 'delete' links on some tables are now actually buttons instead of plaintext links.

File: app/views/config_templates/index.html.erb [hunk 10]
```diff
@@ -17,12 +17,12 @@
       <th><%= sort :kind, :as => _("Kind") %></th>
       <th><%= sort :snippet, :as => s_("ConfigTemplate|Snippet") %></th>
       <th><%= sort :locked, :as => s_("ConfigTemplate|Locked"), :default => "DESC" %></th>
-      <th></th>
+      <th>Actions</th>
```
Comment (domcleal) [inline app/views/config_templates/index.html.erb]: extract the string

**Comment by dLobatog** [comment]
Updated with alerts using patternfly classes instead of default jNotify classes. Thanks everyone for reviewing!

**Comment by ohadlevy** [comment]
@tbrisker can you have another round of review? thanks!

File: app/assets/javascripts/config_template.js [hunk 11]
```diff
@@ -112,9 +112,9 @@ function create_editor(item) {
 function set_fullscreen(){
   $('#main').append($("#editor1"));
   $("#editor1")
-     .height($(window).height()-50)
-     .width($('#content').width())
-     .css('top', 10)
+     .height($(window).height())
+     .width($(window).width())
```
Comment (tbrisker) [inline app/assets/javascripts/config_template.js]: This causes some overflow because of the padding on `#main`. you can add `.css('left', -40)` to amend this or just set to `$('#main').width()` and leave it with the padding on the sides.

File: vendor/assets/stylesheets/multi-select.scss [hunk 12]
```diff
@@ -100,27 +89,17 @@
...
-    color:#ffffff;
-    background-color: #aaa;
-    padding: 4px 10px;
-    -webkit-border-radius: 4px 4px 0 0;
-    -moz-border-radius: 4px 4px 0 0;
-    border-radius: 4px 4px 0 0;
+  padding: 4px 10px;
+  border: 1px solid #d1d1d1;
+  background-clip: padding-box;
+  @include linear-gradient(#f9f9f9,#ededed);
```
Comment (tbrisker) [inline vendor/assets/stylesheets/multi-select.scss:95]: This causes the +/- in the multiselect headers to be almost hidden, you only notice them if you happend to hover over them.

File: app/assets/javascripts/application.js [hunk 13]
```diff
@@ -395,11 +402,27 @@ function spinner_placeholder(text){
...
 
 function notify(item, type) {
-  var options = { type: type, sticky: (type != 'success') };
-  $.jnotify($(item).text(), options);
+  var icon = typeToIcon(type);
+  var options = { classMessage: "alert alert-"+type,
+                  classBackground: "",
+                  type: type,
+                  sticky: (type != 'success') };
+  $.jnotify("</div>" + icon + $(item).text(), options);
```
Comment (tbrisker) [inline app/assets/javascripts/application.js:421]: There is no way to close the notification popup. This means that the right most menus (user&administer) can't be accessed without browsing to a different page first.

File: app/assets/stylesheets/patternfly_and_overrides.scss [hunk 14]
```diff
@@ -218,3 +172,15 @@ body.modal-open {
   color: black;
```
Comment (tbrisker) [inline app/assets/stylesheets/patternfly_and_overrides.scss:188]: Is this override still needed?

**Comment by tbrisker** [comment]
Looking good except for a few comments I added inline.
A few of the tables clip the text too soon (esp. host list), but I already have a bug (#8800) open for that. I think this can be addressed after this is merged, as this issue already existed before patternfly, the change to 100% width only makes it stand out more.

**Comment by dLobatog** [comment]
Aside from the alert not disappearing (not sure why yet!), I think I addressed all of your comments.

**Comment by dLobatog** [comment]
Note for self: https://github.com/theforeman/foreman/pull/2137 remove the :heavy_plus_sign: and substitute it by font-awesome plus.

**Comment by ohadlevy** [comment]
@eLobato yes please :)

File: app/assets/javascripts/application.js [hunk 15]
```diff
@@ -395,11 +402,26 @@ function spinner_placeholder(text){
...
+                  type: type };
+  $.jnotify("</div>" + icon + $(item).text(), options);
   $(item).remove();
 }
 
+function typeToIcon(type) {
+  switch(type)
+  {
+  case 'success':
+    return "<span class='pficon pficon-ok'></span><strong>Success: </strong>";
```
Comment (ohadlevy) [inline app/assets/javascripts/application.js]: i18n?

**Comment by ohadlevy** [comment]
maybe its just me, but i still find the initial font size too small, i always by default zoom in, especially, in tables.

**Comment by ohadlevy** [comment]
additionally, the btn imho are also hard to parse visually, for example:
screenshot from 2015-05-13 10 13 02 image

other examples:

image 

image

**Comment by ohadlevy** [comment]
overall, what would be the selling point of this patch in terms of UI, e.g. its better because... ?

**Comment by ehelms** [comment]
For me the selling points are:

1) Patternfly was built by a group of User Interaction designers that have used their knowledge and training combined with user testing and iterations to refine and add to what Bootstrap provides
2) Built on top of bootstrap so you still get almost all of the bootstrappy goodness
3) Adds more interactions, more templates, and more layouts for building enterprise level software

**Comment by dLobatog** [comment]
Latest changes increase the default font by 1px (it makes a difference) & displays some alerts nicer.

**Comment by mmoll** [comment]
I like the look of it, LGTM :+1:

**Comment by ohadlevy** [comment]
@eLobato rebase please?

**Comment by dLobatog** [comment]
@ohadlevy @ehelms Rebased, I think I should restyle a bunch of things after select2.js is now in develop. Basically the style looks off and the loading spinner when you choose an option in a dropdown (say, a Compute resource in Host#new) isn't where it should be.

File: app/assets/javascripts/application.js [hunk 16]
```diff
@@ -420,11 +427,26 @@ function spinner_placeholder(text){
...
+  case 'warning':
+    return "<span class='pficon-layered'><span class='pficon pficon-warning-triangle'></span><span class='pficon pficon-warning-exclamation'></span></span><strong>Warning: </strong>";
```
Comment (ohadlevy) [inline app/assets/javascripts/application.js]: consider using icon helpers? also strings are not extracted.

**Comment by ohadlevy** [comment]
Search btn is not aligned with input

search-alignment

**Comment by ohadlevy** [comment]
dropdown size imho seems a bit too large, they are nearly equal to the edit text.

dropdown-size

**Comment by ohadlevy** [comment]
did you remove Alert close btn on purpose? I just had a scenario where i needed to follow up on the admin dropdown and was unable to reach it besides navigating first to a different location and back.

**Comment by ohadlevy** [comment]
host show graphs are not aligned, they are always below the properties table.

File: app/views/lookup_keys/index.html.erb [hunk 17]
```diff
@@ -16,9 +16,15 @@
         <td class="display-two-pane"><%= link_to_if_authorized trunc_with_tooltip(lookup_key.key), hash_for_edit_lookup_key_path(:id => lookup_key).merge(:auth_object => lookup_key, :permission => 'edit_external_variables', :authorizer => authorizer) %></td>
         <% puppet_class = lookup_key.param_class %>
         <td class="display-two-pane"><%= link_to_if_authorized trunc_with_tooltip(puppet_class), hash_for_edit_puppetclass_path(:id => puppet_class).merge(:auth_object => puppet_class, :authorizer => @puppetclass_authorizer) if puppet_class %></td>
+<<<<<<< HEAD
```
Comment (ohadlevy) [inline app/views/lookup_keys/index.html.erb]: oops

File: app/views/lookup_keys/index.html.erb [hunk 18]
```diff
@@ -16,9 +16,15 @@
...
         <td><%=h lookup_key.lookup_values_count %></td>
         <td><%= display_delete_if_authorized hash_for_lookup_key_path(:id => lookup_key).merge(:auth_object => lookup_key, :permission => 'destroy_external_variables', :authorizer => authorizer),
           :data => { :confirm => _("Delete %s?") % lookup_key.key } %></td>
+=======
```
Comment (ohadlevy) [inline app/views/lookup_keys/index.html.erb]: oops

File: app/views/media/index.html.erb [hunk 19]
```diff
@@ -4,27 +4,32 @@
...
   <tbody>
-    <% for medium in @media %>
+    <% @media.each do |medium| %>
       <tr>
         <td class="display-two-pane"><%= link_to_if_authorized trunc_with_tooltip(medium), hash_for_edit_medium_path(:id => medium).merge(:auth_object => medium, :authorizer => authorizer) %></td>
         <td><%= medium.path %></td>
         <td><%= lookup_family(medium.os_family) %></td>
         <td><%= medium.operatingsystems.map(&:to_label).to_sentence %></td>
         <td>
+<<<<<<< HEAD
```
Comment (ohadlevy) [inline app/views/media/index.html.erb]: oops

File: app/views/media/index.html.erb [hunk 20]
```diff
@@ -4,27 +4,32 @@
...
           <%= display_delete_if_authorized hash_for_medium_path(:id => medium).
             merge(:auth_object => medium, :authorizer => authorizer),
             :data => { :confirm => _("Delete %s?") % medium.name } %>
+=======
```
Comment (ohadlevy) [inline app/views/media/index.html.erb]: oops

**Comment by ohadlevy** [comment]
I've also noticed that the icon on delete hover does not change correctly.

**Comment by ohadlevy** [comment]
another question, did you move the submit buttons (cancel/submit) to the left on purpose ?

**Comment by ohadlevy** [comment]
when editing a provisioning templates, the size of the name column changes (when switching from index to two-pane edit).

**Comment by ohadlevy** [comment]
overall looks pretty solid, thanks @eLobato!

**Comment by ohadlevy** [comment]
@eLobato does patternfly includes data-tables? (https://www.patternfly.org/patterns/table) maybe we can remove it from our vendors.

File: vendor/assets/stylesheets/multi-select.scss (hunk 12 above)
Comment (dLobatog) [inline vendor/assets/stylesheets/multi-select.scss:95]: fixed - now they're blue

**Comment by dLobatog** [comment]
Thanks for the comments @ohadlevy , I fixed the overly big dropdown button, and the jNotify alert close button works again.

I don't know what you mean by host show graphs not being aligned, the color when you hover on delete (which page exactly?) and submit/cancel buttons moved to the left. I see them the same as in `develop`.

Two-pane changes the size of the name column to `col-md-3` in all pages, but that's happening on `develop` as well, is that a problem to fix?

To do:
- search button ~1px misaligned with respect to search text
- data tables
- review if any other important Patternfly conventions are missing

**Comment by ohadlevy** [comment]
@eLobato regarding the delete on hover, see for example in domains, 
delete_on_hover.
sadly screen capture does not save the mouse, but on over, the icon does not change correctly.

**Comment by dLobatog** [comment]
@ohadlevy I'm thinking you have an old version of the PR, I went table by table to convert the last row into action buttons, in domains particularly it should look like this

**Comment by dLobatog** [comment]
Updated once again, I changed a few pages to use Patternfly's [empty state pattern](https://www.patternfly.org/patterns/empty-state/) (see Domain/Reports/Hosts/Auth sources), and some other more minor improvements. Dashboard and statistics page should look nicer now too.

**Comment by dLobatog** [comment]
Updated, all welcome pages use Patternfly's empty state pattern, except for Media, Hostgroup and Partition Tables. Please advise me on next steps to follow, or anything you think that looks wrong here. I can try to look for more things that are not following Patternfly patterns, but I honestly think it'd be better to merge as it is right now as it's already an improvement (in my humble opinion).

When this gets merged, I think we should start moving assets to Bower as @ehelms recommended a while ago, it's getting slightly out of hand.

**Comment by ohadlevy** [comment]
some of the new buttons are not consistent, e.g. under the new welcome page
of realms new is blue vs green in other places.

TBH the main thing that bothers me is how the badges look bad as icon
indicator (e.g. in  hosts list, reports etc).

@tbrisker @domcleal any opinions?

**Comment by domcleal** [comment]
I don't much care for it.  My only comment is that if the fonts are available as an assets gem then please use that and don't vendor them.

**Comment by ohadlevy** [comment]
@eLobato would you rebase please?

**Comment by dLobatog** [comment]
@ohadlevy Rebased, it should be more or less ready but I'm holding off from making major updates to this PR until Rails 4 is merged - not because of conflicts, just that Rails 4 is more urgent.

**Comment by mmoll** [comment]
It seems there's a test failure... Otherwise I'm still :+1: - if there are still some small things missing, these can get fixed when in develop anyway and it would get much wider exposure.

**Comment by ohadlevy** [comment]
@dLobatog any chance to rebase? :)

File: app/assets/javascripts/editor.js [hunk 21]
```diff
@@ -127,6 +127,37 @@ function create_editor() {
   }
 }
 
+function set_fullscreen(){
```
Comment (ohadlevy) [inline app/assets/javascripts/editor.js:130]: do we still need this?

**Comment by ohadlevy** [comment]
@alongoldboim its rebased and yours :)

**Comment by alongoldboim** [comment]
@ohadlevy @dLobatog , thanks for the rebase :+1:

**Comment by alongoldboim** [comment]
@dLobatog we got a problem with rubocop complaining that the Layout helper Module has too many lines, I can move some of it to application helper, would that be ok by you?

File: app/views/images/index.html.erb [hunk 22]
```diff
@@ -8,7 +8,7 @@
       <th><%= s_("Image|Username") %></th>
       <th><%= s_("Image|Uuid") %></th>
       <th><%= s_("Image|User data") %></th>
-      <th></th>
+      <th><<%= _('Actions') %>/th>
```
Comment (ohadlevy) [inline app/views/images/index.html.erb:11]: nitpick: extra <

**Comment by ohadlevy** [comment]
test connection under ldap generate alert-error class instead of alert-danger

**Comment by ohadlevy** [comment]
parameters (hosts/hostgroups and puppet class params) are in different sizes.

**Comment by ohadlevy** [comment]
inhertied option under host edit form is the wrong size

**Comment by ohadlevy** [comment]
imho consider changing default font-size to 14.

**Comment by dLobatog** [comment]
Taken by @alongoldboim on https://github.com/theforeman/foreman/pull/2943
